src/coreclr/vm/methodtablebuilder.cpp

```diff
@@ -3678,7 +3678,10 @@ VOID    MethodTableBuilder::AllocateWorkingSlotTables()
 
     // Allocate a FieldDesc* for each field
     bmtMFDescs->ppFieldDescList = new (GetStackingAllocator()) FieldDesc*[bmtMetaData->cFields];
-    ZeroMemory(bmtMFDescs->ppFieldDescList, bmtMetaData->cFields * sizeof(FieldDesc *));
+    if(bmtMetaData->cFields!=0)
+    {
+        ZeroMemory(bmtMFDescs->ppFieldDescList, bmtMetaData->cFields * sizeof(FieldDesc *));
```

Comment:
Why is this needed? Zero should be a valid input value.

Reply:
but it throws 🤷‍♂️
`memory access out of bounds at corerun.wasm.emscripten_memset_bulkmem`

At least in the current emscripten version we use. I don't know what's the spec for the underlying bulk instruction.

We could test it with latest emscripten and eventually fix it upstream.

Reply:
I think you're not fixing the right problem. Is the problem the size or the fact that `bmtMFDescs->ppFieldDescList` is `NULL`? My point is the size isn't validated in these APIs, the pointer is validated which means that should be the check. A non-`NULL` pointer is valid to have no bytes zero'd out.

Reply:
> StackingAllocator returns `-1` for zero size.

Ugh.